src/main/java/reposense/commits/model/CommitResult.java

```diff
@@ -11,19 +11,21 @@
     private final String hash;
     private final String messageTitle;
     private final String messageBody;
+    private final String tag;
     private final int insertions;
     private final int deletions;
 
     private final transient Author author;
     private final transient Date time;
 
     public CommitResult(Author author, String hash, Date time, String messageTitle,
-            String messageBody, int insertions, int deletions) {
+            String messageBody, String tag, int insertions, int deletions) {
```

Comment:
I have tried using `Optional<String>` for the `tag` field.

When converting to JSON, it will convert the tag fields as JSON objects instead of a key-value pair.

So with empty tag fields, the tag will also appear as empty JSON objects:
<img width="96" alt="Screen Shot 2019-12-17 at 9 38 57 PM" src="https://user-images.githubusercontent.com/32864116/71000085-a422df00-2115-11ea-9d08-de9e648d9d9a.png">
whereas if we use the current implementation (set string to null), the tag field will only appear in the JSON when it is not null.

Reply:
OK, that's fine.